Project 3: Second Milestone for TIC TAC TOE game.

FdI96 opened the pull request:
In this Milestone I added the logic for the game in the following file:
-game_logic.rb

Adapted the user interface file with the logic game's file
-main.rb

Changed the README.md file
-README.md

FdI96 committed:
adding the game logic

FdI96 committed:
fixing linters

FdI96 committed:
fixing linters and adding instructions

FdI96 committed:
Fixing Linters

mahmoud717 reviewed (changes requested):
### Changes required ♻️ 

**Hello @FdI96**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [x] you should have each lib class in its own file.
- [ ] the player class shouldn't be inheriting the board class. if you want to access the methods and attributes of one of them you have to import the other in the file.
 that's what oop is all about, separating different components from each other so that it's easier to read and debug.

- [ ] there should be no puts or print outside of the main file.
- [x] you must use require_relative instead of require if you want to import files. require is for gems only.
- [ ] when it's a draw, please make sure to print it.

### Optional 🟢 

- [ ] It would be really nice if you added a function to ask the players if they want to play again after a game is finished. 

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

FdI96 committed:
fixing issues...

mahmoud717 reviewed (changes requested):
### Changes required ♻️ 

**Hello @FdI96**
 
you haven't fixed all of the requested issues, please make sure to fix the issues before asking for another code review.
---
### Requested 🔴 

- [ ] the player class shouldn't be inheriting the board class. if you want to access the methods and attributes of one of them you have to import the other in the file.
 that's what oop is all about, separating different components from each other so that it's easier to read and debug.

- [ ] there should be no puts or print outside of the main file.

- [ ] when it's a draw, please make sure to print it.

- [ ] the game should quit if a player didn't enter P to start. it should say unknown command please enter a valid command or something like this.


---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

FdI96 committed:
Fix issues

mke2111 reviewed (changes requested):
# **STATUS: CHANGES REQUIRED 🔴 ⚠️** 

## Hey @FdI96, 👋🏽 
 **Thank you for the good job you got going on. The game runs almost just fine. Just a few changes to make and we'll be right there.**

## REQUIRED CHANGES ⚠️ 

- Kindly implement **all** changes as requested by the previous reviewer. 

- [ ] Ensure that there are no `puts`, `gets.chomp` or `print` outside the `main.rb` file. Printing and getting user input can only be done from the `main.rb` file.

You need to transfer all methods/code that have `print`, `puts`, and `gets.chomp` into the the main file.

**Kindly make all the requested changes and submit another CRR. 👍🏽** 

## Cheers!! 🍹 & Happy Coding.

# Reach me on slack for a chitchat 🆘  **Roy Mukuye**

mke2111 commented on lib/board.rb:
```diff
@@ -0,0 +1,44 @@
+  def updated_board
+    puts "#{board_array[0]} | #{board_array[1]} | #{board_array[2]}"
+    puts '-- --- --'
+    puts "#{board_array[3]} | #{board_array[4]} | #{board_array[5]}"
```
Kindly transfer the printing logic to the main file

mke2111 commented on lib/board.rb:
```diff
@@ -0,0 +1,44 @@
+    win = false
+    combination_array.each { |elem| elem.all?('X') || elem.all?('O') ? win = true : false }
+    if win
+      puts "Player #{player.name} WINS"
```
Kindly transfer the printing logic to the main file

mke2111 commented on lib/player.rb:
```diff
@@ -0,0 +1,32 @@
+  def select_slot(band, turn, board)
+    puts 'Please select one empty space'
+    board.updated_board
+    number = gets.chomp.to_i
```
Kindly transfer the user input logic to the main file

FdI96 committed:
Fixing Issues

jubaan reviewed (changes requested):
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> ⚠️ 🛠️ ⚠️ 🛠️ Changes Required 🛠️ ⚠️ 🛠️ ⚠️ </h1>
<br />
<p><b>"Fail fast, fail often"</b> and don't let this discourage you, because <b>only by failing do we learn to succeed</b>.</p>
<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#congrats"> 💯 Awesome work 👏🏼 👏🏼 </a></li>
    <li><a href="#mandatory"> ⚠️ Things to Improve </a> 🚨 🚨 </li>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>

<h3 id="congrats"> 1. :100: Awesome work</h3>
<details close="close">
<summary> [click to open] Big or little achievements, but achievements at the end. <b>Feel proud that you're learning and improving everyday.</b></summary>
<!-- BEGIN Awesome work -->

- [ ] You're almost done! 

<!-- END Awesome work  -->
</details>

<hr>
<h3 id="mandatory"> 2. ⚠️ Things to Improve (or the To-Do list to get the project approved) </h3> 
<details close="close">
<summary> [click to open] Working on these observations will help you advance and get closer to the goal 🎯, postpone them or do them half-ways and they'll haunt you 👻 👻... FOREVER!! (It will feel like that, trust me). </summary>
<!-- BEGIN Review -->

- [ ] All logic should be in the library directory. Move `updated_board`, `win_or_draw`. `select_slot` to their parent class or module.
- [ ] Only in the main.rb is allowed to use `prints`, and `puts`.
- [ ] Is not just moving the methods, but to take out the prints and puts from the logic and when application makes the call in the fronend, then it prints or outputs the returned value of the method.

<!-- END Review -->
</details>
<p><b>NOTE</b>This list is or could be divided in what is show in this comment and inline comments posted below.<p>
<hr>

<h3 id="optional"> 3. 💡 Nice to have and other comments </h3>
<details close="close">
<summary> [click to open] </h3></summary>
<!-- BEGIN Optional and Comments -->

- [ ] Nothing more to add.

<!-- END Optional and Comments -->
</details>

<h3 id="contact"> 4. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    <img src="https://avatars.githubusercontent.com/u/25774518?s=400&u=ae20ddf2d6fb3cae9e2f4044afe3b93d19e5b69e&v=4" alt="Julio Añoveros" title="@jubaan" width="100px" heigh="100px"/>
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 5. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> <img src="https://www.pinclipart.com/picdir/big/130-1305924_rubber-ducky-clipart-eastern-iowa-duck-race-duck.png" width="18px" height="18px">... it would be great to own one 😄. </p>

<h2 align="center"> Keep up the great work :muscle: and Happy Coding :smiley: :computer:!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

FdI96 committed:
class moved to main.rb

I've moved the class to the main.rb file due to all the method from there uses puts. and gets.chomp

jubaan commented on bin/main.rb:
```diff
@@ -8,108 +9,103 @@ def updated_board(board_array)
-def player_turn(board_array, turn)
-  band = false
-  band = select_slot(band, turn, board_array) until band
+  def win_or_draw(accu, player, combination_array)
+    win = false
+    combination_array.each { |elem| elem.all?('X') || elem.all?('O') ? win = true : false }
+    if win
+      puts "Player #{player.name} WINS"
+      false
+    elsif accu == 9
+      puts 'No more plays available. Its a DRAW'
+      false
+    else
+      true
+    end
+  end
```
- [ ] This logic should be inside the `lib` directory. It also should not contain `puts` nor `prints`.

jubaan commented on bin/main.rb:
```diff
@@ -8,108 +9,103 @@ def updated_board(board_array)
-  if (1..9).include?(number)
-    if board_array[number - 1] == 'X' || board_array[number - 1] == 'O'
-      puts 'Slot occupied'
+  def select_slot(band, turn, board)
+    puts 'Please select one empty space'
+    number = gets.chomp.to_i
+    if (1..9).include?(number)
+      if board.board_array[number - 1] == 'X' || board.board_array[number - 1] == 'O'
+        puts 'Slot occupied'
+      else
+        board.modify_board(number, turn)
+        band = true
+      end
```
- [ ] This logic should be inside the `lib` directory. It also should not contain `puts` nor `prints`.

jubaan commented on bin/main.rb:
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env ruby
+require_relative '../lib/board.rb'
 
 def updated_board(board_array)
```
- [ ] This logic should be inside the `lib` directory. It also should not contain `puts` nor `prints`.

jubaan reviewed (changes requested):
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> ⚠️ 🛠️ ⚠️ 🛠️ Changes Required 🛠️ ⚠️ 🛠️ ⚠️ </h1>
<br />
<p><b>"Fail fast, fail often"</b> and don't let this discourage you, because <b>only by failing do we learn to succeed</b>.</p>
<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#congrats"> 💯 Awesome work 👏🏼 👏🏼 </a></li>
    <li><a href="#mandatory"> ⚠️ Things to Improve </a> 🚨 🚨 </li>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>

<h3 id="congrats"> 1. :100: Awesome work</h3>
<details close="close">
<summary> [click to open] Big or little achievements, but achievements at the end. <b>Feel proud that you're learning and improving everyday.</b></summary>
<!-- BEGIN Awesome work -->

<!-- END Awesome work  -->
</details>

<hr>
<h3 id="mandatory"> 2. ⚠️ Things to Improve (or the To-Do list to get the project approved) </h3> 
<details close="close">
<summary> [click to open] Working on these observations will help you advance and get closer to the goal 🎯, postpone them or do them half-ways and they'll haunt you 👻 👻... FOREVER!! (It will feel like that, trust me). </summary>
<!-- BEGIN Review -->

- [ ] The same issues still persist please review your `main.rb` file.
> * [ ] All logic should be in the library directory. Move `updated_board`, `win_or_draw`. `select_slot` to their parent class or module.
> * [ ]  Only in the main.rb is allowed to use `prints`, and `puts`.
> * [ ]  Is not just moving the methods, but to take out the prints and puts from the logic and when application makes the call in the fronend, then it prints or outputs the returned value of the method.



<!-- END Review -->
</details>
<p><b>NOTE</b>This list is or could be divided in what is show in this comment and inline comments posted below.<p>
<hr>

<h3 id="optional"> 3. 💡 Nice to have and other comments </h3>
<details close="close">
<summary> [click to open] </h3></summary>
<!-- BEGIN Optional and Comments -->

- [ ] Nothing more to add.

<!-- END Optional and Comments -->
</details>

<h3 id="contact"> 4. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    <img src="https://avatars.githubusercontent.com/u/25774518?s=400&u=ae20ddf2d6fb3cae9e2f4044afe3b93d19e5b69e&v=4" alt="Julio Añoveros" title="@jubaan" width="100px" heigh="100px"/>
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 5. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> <img src="https://www.pinclipart.com/picdir/big/130-1305924_rubber-ducky-clipart-eastern-iowa-duck-race-duck.png" width="18px" height="18px">... it would be great to own one 😄. </p>

<h2 align="center"> Keep up the great work :muscle: and Happy Coding :smiley: :computer:!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

FdI96 commented:
Hi there ! I’ve asked you some things on slack.

El El mar, 16 feb 2021 a las 0:11, Julio Añoveros <notifications@github.com>
escribió:

> *@jubaan* requested changes on this pull request.
>
> [image: Microverse] <https://www.microverse.org/?grsf=kfm2ej>
> ⚠️ 🛠️ ⚠️ 🛠️ Changes Required 🛠️ ⚠️ 🛠️ ⚠️
>
> *"Fail fast, fail often"* and don't let this discourage you, because *only
> by failing do we learn to succeed*.
> Review Summary
>
>    1. 💯 Awesome work 👏🏼 👏🏼 <#m_-8243506911071514028_congrats>
>    2. ⚠️ Things to Improve <#m_-8243506911071514028_mandatory> 🚨 🚨
>    3. 💡Nice to have and other comments ✍🏼 ✍🏼
>    <#m_-8243506911071514028_optional>
>    4. 🧔🏼 About your Reviewer <#m_-8243506911071514028_contact> (you
>    don't know it... but we are just automated bots 🤖)
>    5. 🆘 * 😭 Feeling stuck? or Need help*? 🙏🏼 🙏🏼
>    <#m_-8243506911071514028_help>
>
> 1. 💯 Awesome work [click to open] Big or little achievements, but
> achievements at the end. *Feel proud that you're learning and improving
> everyday.*
> ------------------------------
> 2. ⚠️ Things to Improve (or the To-Do list to get the project approved) [click
> to open] Working on these observations will help you advance and get closer
> to the goal 🎯, postpone them or do them half-ways and they'll haunt you
> 👻 👻... FOREVER!! (It will feel like that, trust me).
>
>    - The same issues still persist please review your main.rb file.
>
>
>    - All logic should be in the library directory. Move updated_board,
>    win_or_draw. select_slot to their parent class or module.
>    - Only in the main.rb is allowed to use prints, and puts.
>    - Is not just moving the methods, but to take out the prints and puts
>    from the logic and when application makes the call in the fronend, then it
>    prints or outputs the returned value of the method.
>
> *NOTE*This list is or could be divided in what is show in this comment
> and inline comments posted below.
>
> ------------------------------
> 3. 💡 Nice to have and other comments [click to open]
>
>    - Nothing more to add.
>
> 4. 🧔🏼 About your Reviewer [click to open] Say "hello world!" on Slack:
> @jubaan <https://github.com/jubaan>
> [image: Julio Añoveros]
> <https://avatars.githubusercontent.com/u/25774518?s=400&u=ae20ddf2d6fb3cae9e2f4044afe3b93d19e5b69e&v=4>
>
> Julio Añoveros | @jubaan <https://github.com/jubaan>
>
> *Full-stack Web Developer* and
> 🤖 Code Reviewer at Microverse <https://www.microverse.org/?grsf=kfm2ej>🚀
> 5. 🆘 * 😭 Feeling stuck? or Need help*? 🙏🏼 🙏🏼 [click to open] *Feel
> stuck?*
>
> First breath... just do it. Take 15-20 min away from the computer,
> stretch, meditate, do some exercise, drink water, or do whatever is
> necessary to distract your mind from the obstacle.
>
> *Sometimes you just need to get away to see things with a new perspective.*
> [click to open] *Need help?*
>
> Send me a DM in Microverse <https://www.microverse.org/?grsf=kfm2ej>
> Students Slack, you'll find me as @jubaan <https://github.com/jubaan>. If
> I'm not available there are other Code Reviewers you can contact. They'll
> help as well.
>
> Remember that Microverse <https://www.microverse.org/?grsf=kfm2ej> has
> other channels where you can get an answer to your questions:
>
>    1. The *students channel 🧑‍🎓 * in slack,
>    2. your *teammates 👥 *,
>    3. your *mentor 👨🏼 👩🏼 *, and
>    4. *Student Success 🏫 *.
>
> * DON'T hesitate to ask! *
>
> If all of the above doesn't help... get a rubber duck
> <https://rubberduckdebugging.com/>
> <https://camo.githubusercontent.com/ab550bc7f9d5d729665b2eb66c0890ec7b33d778d50afc08dcf880f4974feb76/68747470733a2f2f7777772e70696e636c69706172742e636f6d2f7069636469722f6269672f3133302d313330353932345f7275626265722d6475636b792d636c69706172742d6561737465726e2d696f77612d6475636b2d726163652d6475636b2e706e67>...
> it would be great to own one 😄.
> Keep up the great work 💪 and Happy Coding 😃 💻!!
> ------------------------------
>
> In bin/main.rb
> <https://github.com/FdI96/Tic-Tac-Toe/pull/6#discussion_r576522316>:
>
> > +  def win_or_draw(accu, player, combination_array)
>
> +    win = false
>
> +    combination_array.each { |elem| elem.all?('X') || elem.all?('O') ? win = true : false }
>
> +    if win
>
> +      puts "Player #{player.name} WINS"
>
> +      false
>
> +    elsif accu == 9
>
> +      puts 'No more plays available. Its a DRAW'
>
> +      false
>
> +    else
>
> +      true
>
> +    end
>
> +  end
>
>
>
>    - This logic should be inside the lib directory. It also should not
>    contain puts nor prints.
>
> ------------------------------
>
> In bin/main.rb
> <https://github.com/FdI96/Tic-Tac-Toe/pull/6#discussion_r576522674>:
>
> > +  def select_slot(band, turn, board)
>
> +    puts 'Please select one empty space'
>
> +    number = gets.chomp.to_i
>
> +    if (1..9).include?(number)
>
> +      if board.board_array[number - 1] == 'X' || board.board_array[number - 1] == 'O'
>
> +        puts 'Slot occupied'
>
> +      else
>
> +        board.modify_board(number, turn)
>
> +        band = true
>
> +      end
>
>
>
>    - This logic should be inside the lib directory. It also should not
>    contain puts nor prints.
>
> ------------------------------
>
> In bin/main.rb
> <https://github.com/FdI96/Tic-Tac-Toe/pull/6#discussion_r576523075>:
>
> > @@ -1,4 +1,5 @@
>
>  #!/usr/bin/env ruby
>
> +require_relative '../lib/board.rb'
>
>
>
>  def updated_board(board_array)
>
>
>
>    - This logic should be inside the lib directory. It also should not
>    contain puts nor prints.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FdI96/Tic-Tac-Toe/pull/6#pullrequestreview-590856791>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APB5RQTRTUMMYXY67GIBGV3S7HO5HANCNFSM4XVIXM3Q>
> .
>

FdI96 committed:
re-factored all the code to just have puts on the main.rb and logic in different files

descholar-ceo reviewed (changes requested):
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @FdI96  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] Move thev`updated_board` method to lib folder
- [x] When there is a win, please display the updated board containing the win move
- [x] validate the names of players, don't allow empty names
## Optional :yellow_circle: 
- [ ] It would be very nice if the game doesn't get terminated after a win/draw, it rather asks players whether they want to play another round or not,
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

FdI96 committed:
fixing issues

rindrajosia commented on lib/tools.rb:
```diff
@@ -0,0 +1,7 @@
+def updated_board(board_array)
+  puts "#{board_array[0]} | #{board_array[1]} | #{board_array[2]}"
+  puts '-- --- --'
+  puts "#{board_array[3]} | #{board_array[4]} | #{board_array[5]}"
+  puts '-- --- --'
+  puts "#{board_array[6]} | #{board_array[7]} | #{board_array[8]}"
+end
```
No `puts/print` and `gets.chomp` outside `main.rb` file. Please, move this in your `main.rb` file.

rindrajosia reviewed (changes requested):
**Required Changes**
Hi, I'm Josia your reviewer for this PR
Good job improving your project! There are still some issues to revise before approving this project but hang in there and read the comments below with care. You can do it! :muscle: 

- [x] [REQUIRED] Your game should not `exit` or `raise an error` if the user enters an empty name, it should ask him `to enter his name again`.
![2](https://user-images.githubusercontent.com/58552748/108032393-756b0600-7043-11eb-9178-da365ef2764f.png)

- [x]  [REQUIRED] Your game should `not raise an error` if the user enters a letter different from 'S' when you ask him to start the game. It should just stop with a message like 'Goodbye'.
![1](https://user-images.githubusercontent.com/58552748/108032422-80be3180-7043-11eb-8a3c-6cca7b253f61.png)

- [x]  [REQUIRED] No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

_This project is almost done!_
Happy coding
Rindra Josia

FdI96 committed:
fixing issues

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello Federico Lopez Cechini** 
Good job with the improvements for this project :muscle:  :muscle:
We still need to make some improvements before you can go ahead:

- [ ] Fix rubocop offenses
- [ ] The instruction says "Press 'S' to start" but I guess you actually mean "Type 'S' to start" (the same for "Press 'P' to start")
- [ ] Validate player1 and player2 have different names
- [ ] In Board class, combination_array is never directly read outside the class so it doesn't need the attr_reader.
- [ ] In the main.rb file, the "if" on line 47 has too many responsibilities, try to split its logic into different functions.
- [ ] In the main.rb file, try to organize your code in function with descriptive names.
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

FdI96 committed:
Fixing Issues and linters

FdI96 commented:
Hey Oscar! I've talked to you on Slack.

El mar, 16 feb 2021 a las 16:19, Oscar Andrés Russi Porras (<
notifications@github.com>) escribió:

> *@andresporras3423* requested changes on this pull request.
> Status: Changes Required ♻️
>
> *Hello Federico Lopez Cechini*
> Good job with the improvements for this project 💪 💪
> We still need to make some improvements before you can go ahead:
>
>    - Fix rubocop offenses
>    - The instruction says "Press 'S' to start" but I guess you actually
>    mean "Type 'S' to start" (the same for "Press 'P' to start")
>    - Validate player1 and player2 have different names
>    - In Board class, combination_array is never directly read outside the
>    class so it doesn't need the attr_reader.
>    - In the main.rb file, the "if" on line 47 has too many
>    responsibilities, try to split its logic into different functions.
>    - In the main.rb file, try to organize your code in function with
>    descriptive names.
>
> Please request another review once you are ready! 👍
>
> Happy coding ☺️
>
> Oscar Andrés Russi Porras
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FdI96/Tic-Tac-Toe/pull/6#pullrequestreview-591547194>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APB5RQWVNTLTW7RENVQJ6WTS7LAKJANCNFSM4XVIXM3Q>
> .
>

kevcap reviewed (approved):
# Great job! Project approved! 💪

Any questions feel free to reach me through slack.
